plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -147,11 +147,20 @@ export type SearchBarProps = Partial<SearchBarBaseProps>;
  * @public
  */
 export const SearchBar = ({ onChange, ...props }: SearchBarProps) => {
+  const configApi = useApi(configApiRef);
+  const analytics = useAnalytics();
   const { term, setTerm } = useSearch();
 
   const handleChange = (newValue: string) => {
-    setTerm(newValue);
-    if (onChange) onChange(newValue);
+    if (configApi.getOptionalConfig('app.analytics.ga')) {
```

Comment:
I think this breaks the abstraction around the analytics API, probably best to unconditionally call the API instead

Reply:
Ah I get it, right now GA is supported but in the future we may see more integrations. Will update!